hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/streaming/HoodieStreamSource.scala

```diff
@@ -186,11 +160,11 @@ class HoodieStreamSource(
     }
   }
 
-  private def startCommitTime(startOffset: HoodieSourceOffset): String = {
+  private def startCompletionTime(startOffset: HoodieSourceOffset): String = {
     startOffset match {
-      case INIT_OFFSET => startOffset.commitTime
-      case HoodieSourceOffset(commitTime) =>
-        commitTime
+      case INIT_OFFSET => startOffset.completionTime
+      case HoodieSourceOffset(completionTime) =>
+        HoodieInstantTimeGenerator.instantTimePlusMillis(completionTime, 1)
```

Comment:
why plus 1 millisecond?

Reply:
This is to skip the open range start based on the last checkpoint of the stream source.  I'm trying to see if there a better and efficient way to get the completion time of next commit.

Reply:
Isn't the begin completion time of current read just the end completion time of last read?

Reply:
Yes in the sense of `OPEN_CLOSED` range type.  However,`CLOSED_CLOSED` range type is used in the Spark incremental relation.

Reply:
I understand your point now.  I've changed the `IncrementalRelation`, `MergeOnReadIncremetnalRelation`, and `CDCRelation` to take `rangeType` as an argument so it can be dynamically changed based on different scenarios.  Now the logic in `HoodieStreamSource` is easy to understand, without using `HoodieInstantTimeGenerator#instantTimePlusMillis`.

Reply:
One question I have is for `CDCRelation`. The default range type is hardcoded to `OPEN_CLOSED` before this PR.  I've maintained that.  Let me know if I need to change it too.